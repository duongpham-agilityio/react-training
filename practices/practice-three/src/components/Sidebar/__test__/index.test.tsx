import { fireEvent, render } from '@testing-library/react';

// Components
import { SideBar } from '..';

const setup = () => render(<SideBar />);

describe('Sidebar', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('Show full sidebar', () => {
    const { getByRole, getByText } = setup();

    fireEvent.click(getByRole('button'));

    expect(getByText('Wishlist')).toBeDefined();
  });
});
