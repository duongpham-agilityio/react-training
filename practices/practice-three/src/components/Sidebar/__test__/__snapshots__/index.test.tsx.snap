// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Match to snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-1733sf5"
  >
    <ul
      class="css-155za0w"
      role="list"
    >
      <li
        class="css-1hnnqja"
      >
        <a
          class="chakra-link css-18nqsgs"
        >
          <svg
            fill="none"
            height="45"
            width="53"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.009 1C7.904 1 1 7.166 1 15.175c0 5.524 2.542 10.175 5.912 14.028 3.357 3.84 7.69 7.063 11.606 9.76l6.757 4.655c.74.51 1.71.51 2.45 0l6.757-4.654c3.917-2.698 8.249-5.921 11.606-9.76C49.458 25.35 52 20.698 52 15.173 52 7.167 45.096 1 36.991 1 32.814 1 29.138 2.993 26.5 5.572 23.862 2.993 20.186 1 16.009 1Z"
              fill="#E86F6F"
              stroke="#FF5E5E"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-opacity="0.5"
            />
          </svg>
        </a>
      </li>
      <li
        class="css-89a4w2"
      >
        <a
          class="chakra-link css-18nqsgs"
        >
          <svg
            fill="none"
            height="51"
            width="53"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M47.45 7.533H11.603L8.54 2.005A1.944 1.944 0 0 0 6.84 1H2.947A1.953 1.953 0 0 0 1 2.96c0 1.082.871 1.96 1.946 1.96h2.75l3.023 5.456 6.972 15.531.007.016.614 1.368-7 7.518a1.97 1.97 0 0 0-.407 2.016 1.95 1.95 0 0 0 1.607 1.272l6.38.734c7.979.918 16.037.918 24.017 0l6.38-.734A1.957 1.957 0 0 0 49 35.927a1.95 1.95 0 0 0-2.155-1.725l-6.38.734a101.17 101.17 0 0 1-23.133 0l-2.531-.291 5.137-5.518c.05-.053.096-.109.138-.166l1.954.256c2.738.359 5.505.43 8.258.21a24.055 24.055 0 0 0 16.523-8.542l1.5-1.79c.05-.06.097-.123.14-.189l2.796-4.29c1.982-3.042-.185-7.083-3.797-7.083ZM14.626 42.16c-2.15 0-3.893 1.755-3.893 3.92S12.474 50 14.625 50c2.15 0 3.893-1.755 3.893-3.92s-1.743-3.92-3.893-3.92ZM39.281 46.08c0-2.165 1.743-3.92 3.893-3.92 2.15 0 3.893 1.755 3.893 3.92S45.324 50 43.174 50c-2.15 0-3.893-1.755-3.893-3.92Z"
              fill="#979797"
            />
            <path
              d="M47.45 7.533H11.603L8.54 2.005A1.944 1.944 0 0 0 6.84 1H2.947A1.953 1.953 0 0 0 1 2.96c0 1.082.871 1.96 1.946 1.96h2.75l3.023 5.456 6.972 15.531.007.016.614 1.368-7 7.518a1.97 1.97 0 0 0-.407 2.016 1.95 1.95 0 0 0 1.607 1.272l6.38.734c7.979.918 16.037.918 24.017 0l6.38-.734A1.957 1.957 0 0 0 49 35.927a1.95 1.95 0 0 0-2.155-1.725l-6.38.734a101.17 101.17 0 0 1-23.133 0l-2.531-.291 5.137-5.518c.05-.053.096-.109.138-.166l1.954.256c2.738.359 5.505.43 8.258.21a24.055 24.055 0 0 0 16.523-8.542l1.5-1.79c.05-.06.097-.123.14-.189l2.796-4.29c1.982-3.042-.185-7.083-3.797-7.083ZM14.626 42.16c-2.15 0-3.893 1.755-3.893 3.92S12.474 50 14.625 50c2.15 0 3.893-1.755 3.893-3.92s-1.743-3.92-3.893-3.92ZM39.281 46.08c0-2.165 1.743-3.92 3.893-3.92 2.15 0 3.893 1.755 3.893 3.92S45.324 50 43.174 50c-2.15 0-3.893-1.755-3.893-3.92Z"
              stroke="#8F8F8F"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-opacity="0.5"
            />
          </svg>
        </a>
      </li>
      <li
        class="css-nxlbid"
      >
        <a
          class="chakra-link css-18nqsgs"
        >
          <svg
            fill="currentColor"
            height="32"
            viewBox="0 0 16 16"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 6a3 3 0 1 1-6 0 3 3 0 0 1 6 0z"
            />
            <path
              d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm8-7a7 7 0 0 0-5.468 11.37C3.242 11.226 4.805 10 8 10s4.757 1.225 5.468 2.37A7 7 0 0 0 8 1z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
      </li>
      <li
        class="css-1q2uw3t"
      >
        <a
          class="chakra-link css-gyjzfl"
        >
          <svg
            fill="none"
            height="40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 0c.77 0 1.395.625 1.395 1.395v1.86a1.395 1.395 0 1 1-2.79 0v-1.86C18.605.625 19.229 0 20 0ZM20 9.302c-5.908 0-10.698 4.79-10.698 10.698S14.092 30.698 20 30.698 30.698 25.908 30.698 20 25.908 9.302 20 9.302ZM7.831 5.858A1.395 1.395 0 1 0 5.858 7.83l1.315 1.316a1.395 1.395 0 0 0 1.974-1.974L7.83 5.858ZM40 20c0 .77-.625 1.395-1.395 1.395h-1.86a1.395 1.395 0 1 1 0-2.79h1.86c.77 0 1.395.624 1.395 1.395ZM34.142 7.831a1.395 1.395 0 0 0-1.973-1.973l-1.316 1.315a1.395 1.395 0 0 0 1.974 1.974l1.315-1.316ZM20 35.349c.77 0 1.395.625 1.395 1.395v1.86a1.395 1.395 0 0 1-2.79 0v-1.86c0-.77.624-1.395 1.395-1.395ZM32.827 30.853a1.395 1.395 0 1 0-1.974 1.974l1.316 1.315a1.395 1.395 0 0 0 1.973-1.973l-1.315-1.316ZM4.651 20c0 .77-.625 1.395-1.395 1.395h-1.86a1.395 1.395 0 1 1 0-2.79h1.86c.77 0 1.395.624 1.395 1.395ZM9.147 32.827a1.395 1.395 0 0 0-1.974-1.974L5.858 32.17a1.395 1.395 0 1 0 1.973 1.973l1.316-1.315Z"
              fill="#565656"
            />
          </svg>
        </a>
      </li>
    </ul>
    <div
      class="css-17c1v76"
    >
      <button
        aria-label="Search database"
        class="chakra-button css-16yye98"
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="#737373"
          focusable="false"
          height="28"
          viewBox="0 0 16 16"
          width="28"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
